### PR TITLE
Implemented HTMLFormElement.relList  

### LINKED ISSUE #27252: Implement HTMLFormElement.relList
Spec:
* https://html.spec.whatwg.org/#the-form-element:dom-form-rellist
* https://html.spec.whatwg.org/#dom-form-rellist

This can be implemented by duplicating the implementation of [HTMLAnchorElement](https://github.com/servo/servo/blob/9fe92ecb4c35747fd58b13f6ae23eb0f31a14680/components/script/dom/htmlanchorelement.rs#L137-L141) as well as adding the missing IDL to `components/script/dom/webidls/HTMLFormElement.webidl`.

Tests: `./mach test-wpt tests/wpt/web-platform-tests/html/dom/idlharness.https.html`
We will need to update `tests/wpt/metadata/html/dom/idlharness.https.html.ini` to remove the expected failures that will now pass after this change.

### PR DESCRIPTION
**Pull request by avr1254**
Updated the tests to reflect addition of rel and relList for HTMLFormElement, as well as porting those code snippets from HTMLAnchorElement.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #27252  (GitHub issue number if applicable)

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Comment by jdm** [comment]
A couple other test results need to be updated:
```
  ▶ Unexpected subtest result in /WebIDL/ecmascript-binding/put-forwards.html:
  └ PASS [expected FAIL] Setting form.relList to noreferrer is reflected in rel
  ▶ Unexpected subtest result in /html/dom/idlharness.https.html?include=HTML.*:
  └ PASS [expected FAIL] HTMLFormElement interface: document.createElement("form") must inherit property "rel" with the proper type
  ▶ Unexpected subtest result in /html/dom/idlharness.https.html?include=HTML.*:
  └ PASS [expected FAIL] HTMLFormElement interface: document.createElement("form") must inherit property "relList" with the proper type
```
That's `tests/wpt/metadata/WebIDL/emcascript-binding/put-forwards.html.ini` and `tests/wpt/metadata/html/dom/idlharness.https.html.ini`.

**Comment by avr1254** [comment]
Thanks! Updating now.

**Comment by avr1254** [comment]
I don't have a clue what went wrong this time. Maybe it's my configuration?

File: components/script/dom/htmlformelement.rs [hunk 1]
```diff
@@ -1189,7 +1208,6 @@ impl HTMLFormElement {
         // Step 9
         Some(form_data.datums())
     }
-
```
Comment (jdm) [inline components/script/dom/htmlformelement.rs:1192]: nit: could you add this newline back?

File: tests/wpt/metadata/WebIDL/ecmascript-binding/put-forwards.html.ini [hunk 2]
```diff
@@ -1,4 +1,3 @@
 [put-forwards.html]
   type: testharness
-  [Setting form.relList to noreferrer is reflected in rel]
-    expected: FAIL
```
Comment (jdm) [inline tests/wpt/metadata/WebIDL/ecmascript-binding/put-forwards.html.ini:4]: This file can be removed now that it doesn't contain any non-pass results.

**Review by jdm — APPROVED** [review]
This looks good! Could you squash all the commits into one?

**Comment by avr1254** [comment]
I'm having a bit of trouble rebasing, since I accidentally updated my local repository with commits on the master here. Any advice?

**Comment by jdm** [comment]
@avr1254 If you can find the last commit you made to the branch in `git reflog`, it might be easiest to do a hard reset to that revision and try the squash again.

**Comment by avr1254** [comment]
Yeah, I think this looks good now. It's only my commits.

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

**Pull request merged**